lib/utils.js

```diff
@@ -50,29 +52,67 @@ function retryDelay(attempt) {
 }
 
 /**
- * Returns a google auth client for the current application and provided
- * scopes using a cached client if one exists.
+ * Returns a google auth client for the current application with provided
+ * scopes and configuration, using a cached client if one exists.
  */
-function getApplicationDefaultAuth(scopes, callback) {
-  scopes.sort();
+function getAuthClient(scopes, config, callback) {
+  if (!config) {
+    config = {};
+  } else if (isFunction(config)) {
+    callback = config;
+    config = {};
```

Comment:
Perhaps I am missing something, but because `getAuthClient()` is not exported, it looks like we could make `config` required and force callers to explicitly pass `null` or `undefined` instead of trying to determine if the second or third argument is a callback.